src/utils/components/withMatchMedia.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { MEDIA_QUERIES } from '../designConstants';
+
+const breakpointNames = Object.keys(MEDIA_QUERIES);
+
+/**
+ * @param {String} breakpoint - name of breakpoint (for example, 'medium')
+ * @returns {String} - state property name (for example, isAtMediumUp)
+ */
+export const getStateNameByBreakpoint = breakpoint => {
+	const capitalizedBp = `${breakpoint.substr(0,1).toUpperCase()}${breakpoint.substr(1)}`;
+	return `isAt${capitalizedBp}Up`;
+};
+
+/**
+ * @param {Array} mediaQueries - list of matchMedia-created MediaQueryList objects
+ * @param {Array} breakpoints - array of breakpoint names that were passed to HOC
+ * @returns {Object} - updated `state` object for `withMatchMedia`
+ */
+export const getUpdatedMediaState = (mediaQueries, breakpoints) => mediaQueries
+	.reduce((state, mq, i) => {
+		state[getStateNameByBreakpoint(breakpoints[i])] = mq.matches;
+		return state;
+	}, {});
+
+
+/**
+ * Provides viewport-aware props to wrapped component.
+ *
+ * @param {React.element} WrappedComponent - the component to wrap
+ * @param {Array} breakpoints - array of breakpoint names to watch
+ */
+export const withMatchMedia = (
+	WrappedComponent
+) => class extends React.Component {
+	/**
+	 * @constructor
+	 * @param {Object} - React element props
+	 */
+	constructor(props) {
+		super(props);
+
+		// map breakpoint names to MediaQueryList objects...
+		this.mediaQueries = breakpointNames
+			.map(bp => window.matchMedia(MEDIA_QUERIES[bp]));
+
+		this.state = {
+			media: getUpdatedMediaState(this.mediaQueries, breakpointNames)
```

Comment:
since `breakpointNames` is set as a module constant, you don't need to pass it as an argument. Also, since you can't set `this.mediaQueries` until `componentDidMount`, you might as well just call `this.handleMediaChange()` directly instead of duplicating the state-setting logic.